widgetssdk/src/main/java/com/glia/widgets/callvisualizer/ActivityWatcherController.kt

```diff
@@ -88,13 +98,13 @@ internal class ActivityWatcherController(
         when (state.mode) {
             MODE_NONE -> watcher.dismissAlertDialog(true)
             MODE_MEDIA_UPGRADE -> watcher.showUpgradeDialog(state as MediaUpgrade)
-            MODE_OVERLAY_PERMISSION -> watcher.showOverlayPermissionsDialog()
-            MODE_START_SCREEN_SHARING -> watcher.showScreenSharingDialog()
+            MODE_OVERLAY_PERMISSION -> Logger.d(TAG, "should not be here")
```

Comment:
This actually requires some investigation I forgot to do, The overlay permission dialog is somehow handled differently and the DialogState is not emitted. I will try to make the changes so it would work the same way, but this will require some investigation first. If this starts to work properly, we might even get rid of the permissionType variable from the PR

Reply:
Completely revamped the dialog logic to be in sync with previous implementation